crypto/evp_extra/p_kem.c

```diff
@@ -145,7 +143,7 @@ static int pkey_kem_decapsulate(EVP_PKEY_CTX *ctx,
   }
 
   KEM_KEY *key = ctx->pkey->pkey.kem_key;
-  if (!key->has_secret_key) {
+  if (key->secret_key == NULL) {
```

Comment:
Should we add a similar safety check to encapsulate?

Reply:
added

Reply:
btw, I'm working on a PR to add a bunch of test cases for failure modes of the API. Will add a test for this particular case as well.